build-support/docker/centos6/Dockerfile

```diff
@@ -4,18 +4,33 @@
 # Use centos6 for compatibility with glibc 2.12.
 FROM centos:6
 
-# Install python 2.7.13, (more) modern gcc, and a JDK.
+# Install a (more) modern gcc, and a JDK.
+# Also install sqlite-devel because python 3 errors on startup if not -- see
+# https://stackoverflow.com/a/39907500/2518889.
 RUN yum -y update
+# TODO: figure out why this needs to be installed first for /usr/bin/scl to work!
 RUN yum install -y centos-release-scl
-RUN yum install -y \
-  devtoolset-7-gcc \
-  devtoolset-7-gcc-c++ \
+RUN yum install -y devtoolset-7-gcc{,-c++} \
   git \
   java-1.8.0-openjdk-devel \
-  python27 \
-  libffi \
-  libffi-devel \
-  openssl-devel
+  libffi{,-devel} \
+  openssl-devel \
+  sqlite-devel
+
+# Install the desired versions of python 2 and 3.
+ARG PYTHON_2_VERSION=2.7.13
+ARG PYTHON_3_VERSION=3.6.8
+# NB: PYENV_ROOT must be set in the environment for `pyenv install` to work, otherwise it will
+# mysteriously fail to find the `install` command.
+ENV PYENV_ROOT /pyenv-docker-build
+RUN mkdir ${PYENV_ROOT}
+RUN git clone https://github.com/pyenv/pyenv ${PYENV_ROOT}
+# Install python 2 and 3, and set python 2 as the default interpreter.
+RUN /usr/bin/scl enable devtoolset-7 -- bash -c '\
+    ${PYENV_ROOT}/bin/pyenv install --keep --skip-existing ${PYTHON_2_VERSION} \
+    && ${PYENV_ROOT}/bin/pyenv install --keep --skip-existing ${PYTHON_3_VERSION} \
+    && ${PYENV_ROOT}/bin/pyenv global ${PYTHON_2_VERSION}'
+ENV PATH "${PYENV_ROOT}/shims:${PATH}"
```

Comment:
Why this instead of using pyenv's path manipulation commands?

Reply:
Because I don't know what I'm doing! Will do.

Reply:
This line is actually necessary, unfortunately. I tried running `pants3-in-ci` with all of this the same except for this `ENV` line, and got no interpreter available https://travis-ci.org/pantsbuild/pants/jobs/479577858#L573.

Note that locally you likely have the line `eval "$(pyenv init -)"` in your `.bashrc`. This will do multiple things for you like setup PATH. Here, we instead manually set up PATH ourselves.

Reply:
`pyenv-installer` for whatever reason doesn't actually modify `~/.bashrc` -- see https://github.com/pantsbuild/pants/blob/ad9c5d78da596b33f864fb326863421c121ee54d/build-support/docker/centos6/Dockerfile#L27. So I've added it to `/root/.bashrc` in `centos6/Dockerfile`, and `~/.bashrc` in `travis_ci/Dockerfile` as per https://github.com/pantsbuild/pants/blob/ad9c5d78da596b33f864fb326863421c121ee54d/build-support/docker/travis_ci/Dockerfile#L10, and then source it in the `.travis.yml` as per https://github.com/pantsbuild/pants/blob/ad9c5d78da596b33f864fb326863421c121ee54d/build-support/travis/travis.yml.mustache#L157-L162.